src/components/sequencer.tsx

```diff
@@ -19,108 +18,84 @@ export interface SequencerState {
   pan: number;
 }
 
-class Sequencer extends React.Component<SequencerProps, SequencerState> {
-  static readonly DEFAULT_VOLUME = 100;
-  static readonly DEFAULT_PAN = 0;
+const Sequencer = (props: SequencerProps) => {
+  const { step, audio, id, numSteps } = props;
+  const DEFAULT_VOLUME = 100;
+  const DEFAULT_PAN = 0;
+  const [steps, setSteps] = React.useState(new Array(numSteps).fill(false));
 
-  constructor(props: SequencerProps) {
-    super(props);
-
-    const steps = new Array(props.numSteps).fill(false);
-
-    this.state = {
-      steps,
-      toPlay: false,
-      prevStep: this.props.step,
-      volume: Sequencer.DEFAULT_VOLUME,
-      pan: Sequencer.DEFAULT_PAN
-    };
-
-    this.toggle = this.toggle.bind(this);
-  }
-
-  toggle = (id: number, activated: boolean) => {
-    let { steps } = this.state;
-    steps[id] = activated;
```

Comment:
this is modifying state directly 😞

Reply:
smh past erik

Reply:
lol ya right bro you're still pulling this. Logic is nice doe, I would've maybe use'd a reverse double heap recursive binary search with memoization to get it done, just sayin